### PR TITLE
CI: Integrate Hadolint for Dockerfiles

### PR DESCRIPTION
**Pull request by padmajabhol**
# Pull Request Template

## Fixed linting issues in Dockerfile by integrating Hadolint.
   Integrates CI pipelines to configure hadolint with dockerfiles.

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Additional Context (Please include any Screenshots/gifs if relevant)

...

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings
- [ ] I have tagged the reviewers in a comment below incase my pull request is ready for a review
- [x] I have signed the commit message to agree to Developer Certificate of Origin (DCO) (to certify that you wrote or otherwise have the right to submit your contribution to the project.) by adding "--signoff" to my git commit command.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -38,3 +37,17 @@ jobs:
...
+          dockerfile: "./rest_api_gcbm/Dockerfile
+            ./rest_api_flint.example/Dockerfile
+            ./local/rest_api_flint.example/Dockerfile
+            ./local/rest_api_gcbm/Dockerfile
+            ./layered/cloud-run/Dockerfile
+            ./layered/compute-engine/Dockerfile
+            ./layered/ingress/Dockerfile"
+
+          error_level: 0
+          output_format: tty
```
Comment (HarshCasper) [inline .github/workflows/ci.yml:53]: This works well. However, this step is a part of the docs linting job. Can you refactor this to work with the previous job for better compatibility?

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Can you add a small note about Hadolint on this doc as well: https://github.com/moja-global/FLINT.Cloud/blob/master/docs/DevelopmentSetup/git_and_github_guide.rst

File: .github/workflows/ci.yml (hunk 1 above)
Comment (padmajabhol) [reply .github/workflows/ci.yml]: I have made the changes.

**Review by iamrajiv — APPROVED** [review]

**Comment by padmajabhol** [comment]
@HarshCasper I have made the changes, is this PR good to go now?

File: docs/DevelopmentSetup/git_and_github_guide.rst [hunk 2]
```diff
@@ -126,3 +126,25 @@ Incase your pull request needs further changes, you can update your pull request
...
+
+To pipe your Dockerfile to docker run :
+::
+
+    docker run --rm -i hadolint/hadolint < Dockerfile
+    
+
+
+
+
```
Comment (HarshCasper) [inline docs/DevelopmentSetup/git_and_github_guide.rst:150]: Can you remove these blank lines?

File: docs/DevelopmentSetup/git_and_github_guide.rst [hunk 3]
```diff
@@ -126,3 +126,25 @@ Incase your pull request needs further changes, you can update your pull request
 ::
 
   git push origin <your-branch-name> --force
+
+About Hadolint
+------------------------
+
+Hadolint is a Dockerfile linter that helps you build best practice Docker images.
```
Comment (HarshCasper) [inline docs/DevelopmentSetup/git_and_github_guide.rst:133]: Can you link to the official Hadolint docs?

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by padmajabhol** [comment]
@HarshCasper I have added the link.

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**